src/metabase/driver/sql_jdbc/sync/describe_database.clj

```diff
@@ -0,0 +1,122 @@
+  schema. Returns a reducible sequence of results."
+  [driver ^DatabaseMetaData metadata ^String schema-or-nil ^String db-name-or-nil]
+  (reify
+    clojure.lang.IReduceInit
+    (reduce [_ rf init]
+      ;; tablePattern "%" = match all tables
+      (with-open [rs (.getTables metadata db-name-or-nil (driver/escape-entity-name-for-metadata driver schema-or-nil) "%"
+                                 (into-array String ["TABLE" "VIEW" "FOREIGN TABLE" "MATERIALIZED VIEW" "EXTERNAL TABLE"]))]
+        (reduce
+         rf
+         init
+         (eduction (map #(select-keys % [:table_name :remarks :table_schem]))
+                   (jdbc/reducible-result-set rs {})))))))
+
+(defn fast-active-tables
+  "Default, fast implementation of `active-tables` best suited for DBs with lots of system tables (like Oracle). Fetch
+  list of schemas, then for each one not in `excluded-schemas`, fetch its Tables, and combine the results.
+
+  This is as much as 15x faster for Databases with lots of system tables than `post-filtered-active-tables` (4 seconds
+  vs 60)."
+  [driver ^Connection conn & [db-name-or-nil]]
+  {:pre [(instance? Connection conn)]}
+  (let [metadata (.getMetaData conn)]
+    (eduction
+     (comp (map (fn [schema]
+                  (db-tables driver metadata schema db-name-or-nil)))
+           cat
+           (filter (fn [{table-schema :table_schem, table-name :table_name}]
+                     (i/have-select-privilege? driver conn table-schema table-name))))
+     (syncable-schemas driver metadata))))
+
+(defmethod i/active-tables :sql-jdbc
+  [driver connection]
+  (fast-active-tables driver connection))
+
+(defn post-filtered-active-tables
+  "Alternative implementation of `active-tables` best suited for DBs with little or no support for schemas. Fetch *all*
+  Tables, then filter out ones whose schema is in `excluded-schemas` Clojure-side."
+  [driver ^Connection conn & [db-name-or-nil]]
+  {:pre [(instance? Connection conn)]}
+  (eduction
+   (filter (let [excluded (i/excluded-schemas driver)]
+             (fn [{table-schema :table_schem, table-name :table_name}]
+               (and (not (contains? excluded table-schema))
+                    (i/have-select-privilege? driver conn table-schema table-name)))))
+   (db-tables driver (.getMetaData conn) nil db-name-or-nil)))
+
+(defn describe-database
+  "Default implementation of `driver/describe-database` for SQL JDBC drivers. Uses JDBC DatabaseMetaData."
+  [driver db-or-id-or-spec]
+  {:tables (with-open [conn (jdbc/get-connection (sql-jdbc.conn/db->pooled-connection-spec db-or-id-or-spec))]
+             ;; try to set the Connection to `READ_UNCOMMITED` if possible, or whatever the next least-locking level
+             ;; is. Not sure how much of a difference that makes since we're not running this inside a transaction,
+             ;; but better safe than sorry
+             (sql-jdbc.execute/set-best-transaction-level! driver conn)
+             (transduce
```

Comment:
This would perhaps be clearer as `(into #{} (map ...) (i/active-tables driver conn))`